molo/core/utils.py

```diff
@@ -21,22 +21,57 @@
 from molo.core.api.constants import KEYS_TO_EXCLUDE
 
 
+def copy_translation_pages(page, new_page):
+    from molo.core.models import Languages, LanguageRelation, PageTranslation
+    # Only copy translations for TranslatablePageMixin
+    if not hasattr(page.specific, 'copy_language'):
+        return 'Not translatable page'
+
+    current_site = page.get_site()
+    destination_site = new_page.get_site()
+    if current_site is not destination_site and (page.depth > 2):
+        page.specific.copy_language(current_site, destination_site)
+    languages = Languages.for_site(destination_site).languages
+    if (languages.filter(is_main_language=True).exists() and
+            not new_page.languages.exists()):
+        LanguageRelation.objects.create(
+            page=new_page,
+            language=languages.filter(
+                is_main_language=True).first())
+
+    for translation in page.translations.all():
+        new_lang = translation.translated_page.specific.copy_language(
+            current_site, destination_site)
+        new_translation = translation.translated_page.copy(
+            to=new_page.get_parent())
+        new_l_rel, _ = LanguageRelation.objects.get_or_create(
```

Comment:
For consistency, let's use the same pattern here to update or create as we did with that other block to avoid IntegrityError